packages/cli/src/lib/FlagshipCodeManager.ts

```diff
@@ -0,0 +1,115 @@
+import {EventEmitter} from 'events';
+
+import ansiAlign from 'ansi-align';
+
+import {
+  FLAGSHIP_CODE_DESCRIPTION,
+  FLAGSHIP_CODE_LOGO,
+  FLAGSHIP_CODE_TITLE,
+} from './constants';
+import {FlagshipCodeLogger} from './FlagshipCodeLogger';
+
+/**
+ * Manages the flagship code and ensures there is only one instance of the manager.
+ * Extends EventEmitter to handle events.
+ *
+ * @example
+ * const manager = FlagshipCodeManager.shared;
+ * manager.addAction(async () => {
+ *   // Your action code here
+ * });
+ * manager.run().then(() => {
+ *   console.log('All actions completed');
+ * });
+ */
+export class FlagshipCodeManager extends EventEmitter {
+  protected actions: Function[] = [];
+  protected numberOfActions: number = 0;
+
+  // The single instance of the FlagshipCodeManager
+  protected static instance: FlagshipCodeManager;
+
+  /**
+   * Initializes the FlagshipCodeManager.
+   * This method is a placeholder for any bootstrap logic.
+   */
+  static bootstrap() {
+    process.stdout.write(FLAGSHIP_CODE_LOGO + '\n');
+    process.stdout.write(
+      ansiAlign([FLAGSHIP_CODE_TITLE, FLAGSHIP_CODE_DESCRIPTION]).join('\n') +
+        '\n\n',
+    );
```

Comment:
this needs to be part of the Status components when logger is paused due to `react-ink` clearing out terminal history - this will need to be re-thought.